### PR TITLE
Refresh search favourites

### PR DESCRIPTION
**Pull request by owennw**

### CODE REVIEW

File: src/sidebars/search/search-controller.js [hunk 1]
```diff
@@ -62,8 +62,12 @@
...
                 $scope.$on('updateFavourites', function(event, data) {
                     var index = self.stocks.map(function(stock) { return stock.code; }).indexOf(data.code);
-                    if (index > -1 && !self.query) {
-                        self.stocks.splice(index, 1);
+                    if (index > -1) {
+                        if (!self.query) {
+                            self.stocks.splice(index, 1);
+                        } else {
+                            self.stocks[index].favourite = false;
+                        }
```
Comment (bjedrzejewski) [inline src/sidebars/search/search-controller.js]: Can we have some code comment here? What this block does? I know there was no comment before, but it is becoming bit complicated.

Comment (owennw) [inline src/sidebars/search/search-controller.js]: Sure.

**Comment by owennw** [comment]
Comments added.

**Comment by bjedrzejewski** [comment]
@owennw bug- you cant add favourites

**Comment by bjedrzejewski** [comment]
Double clicking on favorite makes it add twice

**Comment by owennw** [comment]
Can you try retesting now? Make sure your local resource is in a valid state before starting though.

**Comment by bjedrzejewski** [comment]
If you search for a stock, add to fav, select it from search bar, unfav from favbar- it is still selected in search. I know. lol

**Comment by owennw** [comment]
Don't believe ;)

**Comment by bjedrzejewski** [comment]
@owennw yeah it seems that you are right. I will test a bit more and merge. I hope its bullet proof now

**Pull request merged**